### PR TITLE
replace admin addon manage with inline formsets

### LINKED ISSUE #8410: Replace add-on admin with django admin
The current add-on admin is a [custom view](https://github.com/mozilla/addons-server/blob/db3cbaa2e24f132ac5966469402ff4d127c725f4/src/olympia/zadmin/views.py#L354) that re-uses some styles of the django admin, and provides minimal functionality: it lists the add-on name, slug and status, as well as all files attached to a version attached to the add-on. It lets you modify the status of either the add-on or the files.

We should make the django admin provide the same functionality in the django admin change view for add-ons. That would ease integration with other tools based on the django admin like the user admin being built in #7268. It would also allow us to use the same permission checks we're implementing in the rest of the django admin code.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #8410 - replicates the admin manage functionality but doesn't remove the old tool (yet).  The aim is the be feature comparable in this mvp.

### CODE REVIEW

**Comment by eviljeff** [comment]
Screenshot 2020-01-06 20 31 45

File: src/olympia/addons/admin.py [hunk 1]
```diff
@@ -36,19 +44,62 @@ def user_profile_link(self, obj):
...
+    def get_formset(self, request, obj=None, **kwargs):
+        Formset = modelformset_factory(
+            File,
+            form=FileStatusForm,
+            formset=AdminBaseFileFormSet,
+            extra=self.get_extra(request, obj, **kwargs),
+            min_num=self.get_min_num(request, obj, **kwargs),
+            max_num=self.get_max_num(request, obj, **kwargs),
+        )
+        Formset.request = request
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/addons/admin.py]: (Not a blocker, just wondering if you had thought about this)

[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/admin.py]: I think as we're generating a new class each request with `modelformset_factory` we're okay.  (I was working on an alternate implementation for a while that had the pagination in `FileInline` so it didn't need to pass on the request, but I couldn't see how to expose the paginator to the template so I moved it to the formset class)

File: static/js/admin/recalc_hash.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+django.jQuery(document).ready(function ($){
+    "use strict";
+
+    $(document).ajaxSend(function(event, xhr, ajaxSettings) {
+        var csrf;
+        // Block anything that starts with 'http:', 'https:', '://' or '//'.
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply static/js/admin/recalc_hash.js]: it was an improvement over the original (that didn't work with errors), heh, but yeah, I didn't completely rewrite it.  I'll rework.

[the remark this answers is not in the document]
Comment (eviljeff) [reply static/js/admin/recalc_hash.js]: https://github.com/mozilla/addons-server/pull/13185/commits/2ea23240a9dec18803d77d466748b4f2032adb60

File: src/olympia/addons/tests/test_admin.py [hunk 3]
```diff
@@ -284,40 +305,147 @@ def test_can_not_edit_addonuser_if_doesnt_have_admin_advanced(self):
...
+        user = user_factory()
+        self.detail_url = reverse(
+            'admin:addons_addon_change', args=(addon.pk,))
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Addons:Edit')
+        self.grant_permission(user, 'Admin:Advanced')
+        self.client.login(email=user.email)
+        with self.assertNumQueries(35):
+            # It's very high because most of AddonAdmin is unoptimized but we
+            # don't want it unexpectedly increasing.
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/tests/test_admin.py]: https://github.com/mozilla/addons-server/pull/13185/commits/2ea23240a9dec18803d77d466748b4f2032adb60

File: src/olympia/addons/admin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/admin.py]: ... and just after I explain why I can't do it that way I think of something and can do it that way https://github.com/mozilla/addons-server/pull/13185/commits/cf583f33ee191848ecff988c71c3ab8f47e2091a (though we have to set `self.instance` instead, which doesn't seem as bad)

**Pull request merged**